website/docs/reference/widgets/table/inline-editing.md

```diff
@@ -235,13 +235,117 @@ This property contains the details of the row that triggered the action (`onSubm
+| **onSave**    | Action that gets triggered when the user clicks the save button for a cell or a new row.                                                       |
+| **onDiscard** | Action that gets triggered when user clicks discard button for a cell or a new row.                                                       |
+
+## Adding new rows
+
+In addition to editing individual cells, you may allow users to use the Table UI to submit new rows of data. This works similarly to the Inline Editing features.
+
+### How to add a new row
+
+1. Select your Table widget, and ensure that at least one of its columns has its **Editable** property turned on.
+  - Even in new rows, only **Editable** cells can accept user input.
+2. Next, find the **Allow adding a row** property and turn it on as well.
+3. A new button should appear in the Table's UI called 'Add new row'. Click this button to add a row of new cells to the top of the table.
+  - If the Table's **Default Values** property has data added to it, the new cells will be pre-populated with those default values. Otherwise, they will be blank.
+4. Once the cells have been filled in with desired values, clicking the "Save" button will close the editing mode and execute the Table's **onSave** event. Or, clicking "Discard" will throw away the new cells and trigger the **onDiscard** event.
+
+### Capturing and using data from new rows
+
+Saving new rows works much like submitting a form. In order for the data to be sent to update the same datasource populating the table, you must use the Table's **onSave** event to execute a query or function that sends the new data to the right place.
+
+<VideoEmbed host="youtube" videoId="09pOAgK9mhk" title="Using newRow to save new table data" caption="Using newRow to save new table data"/>
+
+Imagine that your table gets information from a query called `myAPI_get`, which is a GET request to a datasource called `myAPI`.
+
+```javascript
+// In the Table Data field
+{{myAPI.data}}
+
+// Returns [{fruit: "pomegranate", color: "red"}, ... ]
+```
+
+And perhaps you add a new row to the table with the data:
+
+```javascript
+{fruit: "banana", color: "yellow"}
+```
+
+To update the original datasource with the new data, you might execute a query like `myAPI_post`, which would be a POST request to `myAPI`. To add the data to the POST request, you can use the Table's `newRow` attribute to reference the new row.
+
+```javascript
+// In our `myAPI_post` query:
+{{Table1.newRow}}
+
+// In the Table's `onSave` event:
+{{myAPI_post.run()}}
+```
+
+Assuming that the POST request is successful, the `myAPI` datasource receives the new information, and the Table should contain the new row the next time it updates.
+
+### Properties
+
+The following properties are related to adding new rows:
+
+| Property              | Type        | Definition               | Code Snippet        |
```

Comment:
Do we have this Code Snippet column in Properties of other widgets. I'm not sure we should be adding this. Relevant examples when explaining the property in detail should suffice.

Reply:
I've included it to match our [Widget Doc Template](https://www.notion.so/appsmith/Widget-Doc-template-5d7fa61a2e2a4ee888487ba9b43bf487) that all of the (updated) widgets use (there are still some needing updated).

I agree though, I don't think the "Code Snippet" column really adds much to any of the tables. All of the binding properties are accessed in a very consistent way, with `<widget_name>.<property_name>`, and it feels like the Code Snippet column just takes up extra space on the page and makes the table a little squished.

Reply:
> All of the binding properties are accessed in a very consistent way, with <widget_name>.<property_name>

(Of course, the moment I said that, I realize that on *this very page* there is an example of a property that does not follow this pattern (`isNewRow`) 😂 . But, I still have the same opinion.)